### PR TITLE
feat: refresh variables when switching service

### LINKED ISSUE #169: Refresh variables when switching pages
For now, variables are only fetched all at once when loading the app (either after login or when refreshing). We need to update corresponding sub-part of the slice when navigating through services and components.

### PR DESCRIPTION
**Pull request by PaulFarault**
Fixes #169

#### Which issue(s) this PR fixes

Fixes #169

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by mdrutel** [comment]
Use Case :
* Click on the "spark" service
* Modify the spark_version attribute
* Click on "Validate" button
* Click on another service, eg. hadoop
* Click on the "spark" service

Result : the attribute is not modified

**Comment by mdrutel** [comment]
addendum : when i refresh another time, the modification is OK ^^'

**Comment by PaulFarault** [comment]
You're right, the value is well actualized in the store but not showed. I'll investigate 👀

**Comment by PaulFarault** [comment]
TL;DR @mdrutel this should work as expected now

The problem was that we are using uncontrolled form. When loading the page, the field is initialized using the `defaultValue` which is not updated along with the state, even if the store is refreshed.

Ideally, we could use controlled form to update the field. This could be worked along with #164.

My solution was to initialize the `defaultValue` with right value since the beginning. I did it by re-fetch the value from the server directly after a successful PUT operation.

**Comment by pvalentin01** [comment]
Use case #1 :
- Click on Services "spark3"
- Modify spark_version attribute
- Click on "Validate" button
- Click on "View" button

Result : The attribute is modified

Use case #2 :
- Click on Services "spark3"
- Modify spark_version attribute
- Click on "Validate" button
- Click on another service
- Click on Services "spark3"

Result : The attribute is modified

**Comment by mdrutel** [comment]
Now it's all fine, variable values are well refreshed ^^

**Review by mdrutel — APPROVED** [review]
The test is OK

**Review by pvalentin01 — APPROVED** [review]
Tests ok
